.github/workflows/perl-cpan-libraries.yml

```
@@ -524,8 +552,165 @@ jobs:
+          # Import Centreon GPG key
+          GPG_KEY_URL="https://yum-gpg.centreon.com/RPM-GPG-KEY-CES"
+          curl -sSL $GPG_KEY_URL -o RPM-GPG-KEY-CES
+          rpm --import RPM-GPG-KEY-CES
+        shell: bash
+
+      - if: ${{ matrix.package_extension == 'deb' }}
+        name: Install zstd, perl and Centreon repositories
+        run: |
+          export DEBIAN_FRONTEND=noninteractive
+          apt-get update
+          apt-get install -y zstd perl wget gpg apt-utils procps
+          wget -O- https://apt-key.centreon.com | gpg --dearmor | tee /etc/apt/trusted.gpg.d/centreon.gpg > /dev/null 2>&1
+          apt-get update
+        shell: bash
+
+      - name: Restore packages from cache
+        uses: actions/cache/restore@6849a6489940f00c2f30c0fb92c6274307ccb58a # v4.1.2
+        with:
+          path: ./*.${{ matrix.package_extension }}
+          key: ${{ github.sha }}-${{ github.run_id }}-${{ matrix.package_extension }}-${{ matrix.distrib }}
+          fail-on-cache-miss: true
+
+      - if: ${{ matrix.package_extension == 'rpm' }}
+        name: Install packages
+        run: |
+          error_log="install_error_${{ matrix.distrib }}_${{ matrix.arch }}.log"
+          for package in ./*.rpm; do
+            echo "Installing package: $package"
+            # List dependencies, and remove version and comparison operators
+            dependencies=$(rpm -qpR $package | sed 's/ [0-9.-]*\(\s\|$\)/ /g' | sed 's/ [<>!=]*\(\s\|$\)/ /g')
+            for dependency in $dependencies; do
+              # Skip non-perl dependencies
+              if [[ $dependency != perl* ]]; then
+                continue
+              else
+                echo "Check dependency: $dependency"
+                # Update the dependency name to match the package name
+                dependency=$(echo $dependency | sed 's/(/-/g' | sed 's/)//g' | sed 's/::/-/g')
+              fi
+              # If the dependency has been built in the same workflow, install it
+              if [[ -n $(find . -maxdepth 1 -regex "\.\/$dependency-[0-9v].*\.rpm") ]]; then
+                echo "Installing dependency: $dependency"
+                error_output=$(dnf install -y ./$dependency*.rpm 2>&1) || { echo "$error_output" >> $error_log; echo "Error during installation of the dependency $dependency" >> $error_log; true; }
+              fi
+            done
+            # Install package, then uninstall it with all his dependencies
+            echo "Package installation..."
+            error_output=$(dnf install -y $package 2>&1) || { echo "$error_output" >> $error_log; echo "Error during installation of the package $package" >> $error_log; true; }
+            echo "Package installation done."
+            echo "Package uninstallation..."
+            error_output=$(dnf autoremove -y $(echo $package | sed 's/_[0-9].*\.rpm//' | sed 's/.\///') 2>&1) || { echo "$error_output" >> $error_log; echo "Error during autoremove of the package $package" >> $error_log; true; }
```

Avoid to download dependencies for each package installation
```suggestion
            error_output=$(dnf autoremove --setopt=keepcache=True -y $(echo $package | sed 's/_[0-9].*\.rpm//' | sed 's/.\///') 2>&1) || { echo "$error_output" >> $error_log; echo "Error during autoremove of the package $package" >> $error_log; true; }
```